sdk/servicebus/azure-servicebus/api_review.py

```diff
@@ -0,0 +1,588 @@
+    entity_name="queue_name",
+    ServiceBusSharedKeyCredential("policy", "key")
+)
+
+# Queue Receiver
+queue_receiver = SeviceBusReceiverClient.from_queue(
+    fully_qualified_namepsace="fully_qualified_namepsace",
+    queue_name="queue_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+queue_receiver = SeviceBusReceiverClient.from_connection_string(
+    conn_str="conn_str",
+    entity_name="queue_name"
+)
+
+queue_receiver = SeviceBusReceiverClient(
+    fully_qualified_namepsace="fully_qualified_namepsace",
+    entity_name="queue_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key"),
+    session="test_session"
+)
+
+# Topic Sender
+topic_sender = SeviceBusSenderClient.from_topic(
+    fully_qualified_namepsace="fully_qualified_namepsace",
+    topic_name="topic_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+topic_sender = SeviceBusSenderClient.from_connection_string(
+    conn_str="conn_str",
+    entity_name="topic_name"
+)
+
+topic_sender = SeviceBusSenderClient(
+    fully_qualified_namepsace="fully_qualified_namepsace",
+    entity_name="topic_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+# Subscription Receiver
+subscription_receiver = SeviceBusReceiverClient.from_topic_subscription(
+    fully_qualified_namepsace="fully_qualified_namespace",
+    topic_name="topic_name",
+    subscription_name="subscription_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+subscription_receiver = SeviceBusReceiverClient.from_connection_string(
+    conn_str="conn_str",
+    entity_name="topic_name",
+    subscription_name="subscription_name"
+)
+
+subscription_receiver = SeviceBusReceiverClient(
+    fully_qualified_namepsace="fully_qualified_namespace",
+    entity_name="topic_name",
+    subscription_name="subcription_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key"),
+    session="test_session"
+)
+
+
+### Send and receive
+# 1 Send
+
+queue_sender = SeviceBusSenderClient.from_queue(
+    fully_qualified_namepsace="fully_qualified_namepsace",
+    queue_name="queue_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+'''
+topic_sender = SeviceBusSenderClient.from_queue(
+    fully_qualified_namepsace="fully_qualified_namepsace",
+    topic_name="topic_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+'''
+
+with queue_sender:
+    # 1.1 send single
+    message = Message("Test message")
+    queue_sender.send(message)
+
+    # 1.2 send list
+    messages = [Message("Test message {}".format(i)) for i in range(10)]
+    queue_sender.send(messages)
+
+    # 1.3 schedule
+    schedule_message = Message("scheduled_message")
+    enqueue_time = datetime.utcnow() + timedelta(minutes=10)
+    sequence_number = queue_sender.schedule(schedule_message, enqueue_time)
+
+    # 1.4 cancel schedule
+    queue_sender.cancel_scheduled_messages(*sequence_number)
```

Comment:
We should be consistent about the singular/list support in these operations:
- `send()`: Seems to support either a single message or a list
- `cancel_scheduled_messages()` supports *args - maybe this should take singular or list as well?
- `schedule()` supports just a single message

Reply:
`List` would introduce the problem of partial success.
What I'm thinking about is:

- `send()`: support sending single `Message` or `BatchMessage` like we did in EventHub before.
- `cancel_scheduled_messages()`: This one should be easy, it takes sequence_numbers, so singular, list would be easy to adapt.
- `schedule`: like `send`, single `Message` and `BatchMessage`.

What's your thought?